### PR TITLE
Test drop

### PR DESCRIPTION
**Pull request by matthiasto**
Add the step option to generate topography
Add parabola and hockey with a step/drop and three release areas
Enable the filehandler to treat empty fields in the ini file (for example resType)
Make Aimec work for multiple release scenarios in the runComparisonModules.py

### CODE REVIEW

File: avaframe/com1DFA/com1DFA.py [hunk 1]
```diff
@@ -495,6 +496,36 @@ def saveInitialParticleDistribution(avaDir, simName, dem):
     savePartToPickle(particles, partDit)
 
 
+def saveMesh(avaDir, simName, dem):
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:499]: please add a quick docstring

File: avaframe/com1DFAPy/DFAtools.py [hunk 2]
```diff
@@ -300,7 +300,7 @@ def getAreaMesh(Nx, Ny, Nz, csz, num):
         _, _, NzNormed = normalize(Nx, Ny, Nz)
         A = csz * csz / NzNormed
     # limit maximum area (for very steeps cells)
-    A = np.where(A > 3*csz*csz, 3*csz*csz, A)
+    # A = np.where(A > 3*csz*csz, 3*csz*csz, A)
```
Comment (fso42) [inline avaframe/com1DFAPy/DFAtools.py:303]: Do you really want to put this change in the master already?

File: avaframe/com1DFAPy/com1DFA.py [hunk 3]
```diff
@@ -1626,6 +1641,30 @@ def readPartFromPickle(inDir, flagAvaDir=False):
...
+        flagAvaDir: bool
+            if True inDir corresponds to an avalanche directory and pickles are
+            read from avaDir/Outputs/com1DFAPy/particles
+    """
+
+    inDir = os.path.join(inDir, 'Mesh')
+    inDir = inDir.replace('Py', '')
+
+    # search for all pickles within directory
+    Mesh = sorted(glob.glob(os.path.join(inDir, '*', '*.p')))
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:1660]: I suggest switching this to `pathlib` with eg. `Path.glob`

File: avaframe/in3Utils/generateTopo.py [hunk 4]
```diff
@@ -118,6 +120,36 @@ def inclinedplane(cfg):
     return x, y, zv
 
 
+def addDrop(cfg, x, y, zv):
+    """ Compute coordinates of inclined plane with given slope (meanAlpha)
```
Comment (fso42) [inline avaframe/in3Utils/generateTopo.py:124]: Please put a more extensive parameter description in the docstring

File: avaframe/in3Utils/generateTopo.py [hunk 5]
```diff
@@ -118,6 +120,36 @@ def inclinedplane(cfg):
...
+    xStartDrop = float(cfg['STEP']['xStartDrop'])
+    dxDrop = float(cfg['STEP']['dxDrop'])
+    alphaDrop = float(cfg['STEP']['alphaDrop'])
+
+    # get zcoord
+    dzDrop = dxDrop * np.tan(np.radians(alphaDrop))
+    xEndDrop = xStartDrop + dxDrop
+
+    nrows, ncols = np.shape(x)
+    zIniDrop, _ = geoTrans.projectOnGrid(xStartDrop*np.ones((nrows)), y[:, 0], np.vstack((zv[0,:], zv)), csz=dx, xllc=x[0, 0], yllc=y[0, 0], interp='bilinear')
```
Comment (fso42) [inline avaframe/in3Utils/generateTopo.py:139]:
```suggestion
    zIniDrop, _ = geoTrans.projectOnGrid(xStartDrop*np.ones((nrows)), 
                                                                y[:, 0], np.vstack((zv[0,:], zv)), 
                                                                csz=dx, xllc=x[0, 0], yllc=y[0, 0], 
                                                                interp='bilinear')
```

File: avaframe/in3Utils/generateTopo.py [hunk 6]
```diff
@@ -118,6 +120,36 @@ def inclinedplane(cfg):
...
+    zIniDrop = np.tile(zIniDrop, (ncols, 1)).transpose()
+    zEndDrop, _ = geoTrans.projectOnGrid(xEndDrop*np.ones((nrows)), y[:, 0], np.vstack((zv[0,:], zv)), csz=dx, xllc=x[0, 0], yllc=y[0, 0], interp='bilinear')
```
Comment (fso42) [inline avaframe/in3Utils/generateTopo.py:141]: see above

**Review by fso42 — CHANGES_REQUESTED** [review]
Those .keep files are redundant, since the folder is not empty. So please remove them.

File: avaframe/com1DFAPy/DFAfunctionsCython.pyx [hunk 7]
```diff
@@ -582,9 +587,9 @@ def updatePositionC(cfg, particles, dem, force):
 
     if uMag > 0.0:
       # ensure that velocitity magnitude stays the same also after reprojection onto terrain
-      uxNew = uxNew * uMag / (uMagNew + velMagMin)
-      uyNew = uyNew * uMag / (uMagNew + velMagMin)
-      uzNew = uzNew * uMag / (uMagNew + velMagMin)
+      uxNew = uxNew * uMag / (uMagNew + 0*velMagMin)
+      uyNew = uyNew * uMag / (uMagNew + 0*velMagMin)
+      uzNew = uzNew * uMag / (uMagNew + 0*velMagMin)
```
Comment (awirb) [inline avaframe/com1DFAPy/DFAfunctionsCython.pyx:592]: is this on purpose? - or can 0*velMagMin be removed?

File: avaframe/in3Utils/generateTopoCfg.ini [hunk 8]
```diff
@@ -101,6 +103,17 @@ damWidth = 75
 
 damHeight = 75
 
+# Add a drop in the topography (in x direction)
+[STEP]
+# slope angle of the step in the inclined plane with step [°]
+# this parameter required for SP
```
Comment (awirb) [inline avaframe/in3Utils/generateTopoCfg.ini:109]: SP?

**Review by awirb — CHANGES_REQUESTED** [review]

**Comment by matthiasto** [comment]
to be continued...

**Pull request closed without merging**